src/components/QueryHeader.tsx

```diff
@@ -0,0 +1,206 @@
+import { SelectableValue } from '@grafana/data';
+import { EditorField, EditorHeader, EditorRow, FlexItem, InlineSelect, Space } from '@grafana/experimental';
+import { Button, ConfirmModal, InlineSwitch, RadioButtonGroup, Select } from '@grafana/ui';
+import { BigQueryAPI } from 'api';
+import React, { useCallback, useState } from 'react';
+import { DEFAULT_REGION, PROCESSING_LOCATIONS, QUERY_FORMAT_OPTIONS } from '../constants';
+import { BigQueryQueryNG, EditorMode, QueryFormat, QueryRowFilter, QueryWithDefaults } from '../types';
+import { DatasetSelector } from './DatasetSelector';
+import { TableSelector } from './TableSelector';
+
+interface QueryHeaderProps {
+  query: QueryWithDefaults;
+  onChange: (query: BigQueryQueryNG) => void;
+  onRunQuery: () => void;
+  onQueryRowChange: (queryRowFilter: QueryRowFilter) => void;
+  queryRowFilter: QueryRowFilter;
+  sqlCodeEditorIsDirty: boolean;
+  apiClient: BigQueryAPI;
+}
+
+const editorModes = [
+  { label: 'Builder', value: EditorMode.Builder },
```

Comment:
EditorMode enum should eventually be moved to experimental so that we don't constantly re-define it in plugins